src/I18n/I18n.php

```diff
@@ -70,7 +71,11 @@ public static function translators()
         );
 
         if (class_exists('Cake\Cache\Cache')) {
-            static::$_collection->setCacher(Cache::engine('_cake_core_'));
+            try {
+                static::$_collection->setCacher(Cache::engine('_cake_i18n_'));
+            } catch (InvalidArgumentException $e) {
+                static::$_collection->setCacher(Cache::engine('_cake_core_'));
+            }
```

Comment:
Clever way to get backwards compatibility and let people who need the separate cache use it.